dpnp/backend/kernels/dpnp_krnl_manipulation.cpp

```diff
@@ -53,24 +54,16 @@ void dpnp_elemwise_transpose_c(void* array1_in,
     _DataType* array1 = reinterpret_cast<_DataType*>(array1_in);
     _DataType* result = reinterpret_cast<_DataType*>(result1);
 
-    const size_t input_shape_size = input_shape.size();
-    size_t* input_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(input_shape_size * sizeof(long)));
-    size_t* result_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(input_shape_size * sizeof(long)));
+    size_t* input_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(ndim * sizeof(long)));
+    get_shape_offsets_inkernel(input_shape, ndim, input_offset_shape);
 
-    size_t dim_prod_input = 1;
-    size_t dim_prod_result = 1;
-    for (long i = input_shape_size - 1; i >= 0; --i)
+    size_t* temp_result_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(ndim * sizeof(long)));
+    get_shape_offsets_inkernel(result_shape, ndim, temp_result_offset_shape);
+
+    size_t* result_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(ndim * sizeof(long)));
+    for (size_t axis = 0; axis < ndim; ++axis)
```

Comment:
Don't we want to parallelize the loop with sycl?

Reply:
Of course if it's possible.

Reply:
this loop over shape (not data). I don't think it will bring more performance but gives outstanding complexity.

Reply:
I agree that the loop is too short to parallelize.